mmdet/apis/test.py

```diff
@@ -81,17 +117,28 @@ def multi_gpu_test(model, data_loader, tmpdir=None, gpu_collect=False):
     rank, world_size = get_dist_info()
     if rank == 0:
         prog_bar = mmcv.ProgressBar(len(dataset))
+
     for i, data in enumerate(data_loader):
+
+        torch.cuda.synchronize()
+        start_time = time.perf_counter()
+
         with torch.no_grad():
             result = model(return_loss=False, rescale=True, **data)
-        results.append(result)
 
         if rank == 0:
-            batch_size = (
-                len(data['img_meta']._data)
-                if 'img_meta' in data else data['img'][0].size(0))
-            for _ in range(batch_size * world_size):
-                prog_bar.update()
+            torch.cuda.synchronize()
+            elapsed = time.perf_counter() - start_time
+            batch_size = data['img'][0].size(0)
```

Comment:
We need to keep using to obtain the `batch_size` in case downstream repos' inputs do not have the key `img`.
```python
batch_size = (
                len(data['img_meta']._data)
                if 'img_meta' in data else data['img'][0].size(0))
```